### PR TITLE
Support hooking middie in multiple hooks

### PR DESCRIPTION
**Pull request by mcollina**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by kibertoad — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -92,6 +94,35 @@ async function subsystem (fastify, opts) {
...
+If you want to change the Fastify hook that the middleware will be attached to, pass a `hook` option like so:
+
+```js
+const fastify = require('fastify')()
+
+fastify
+  .register(require('middie'), { hook: 'preHandler' })
+  .register(subsystem)
+
+async function subsystem (fastify, opts) {
```
Comment (jsumners) [inline README.md:106]: This is a regular Fastify plugin issuing things during the regular life cycle. Given that this example is about `middie`, there should be something that shows a "middleware" now executing during Fastify's `preHandler` phase.

Comment (mcollina) [reply README.md:106]: As you can see in the example, the plugin call 'use', i.e. middie.

File: test/basic.test.js [hunk 2]
```diff
@@ -625,3 +625,31 @@ test('Should support plugin level prefix', t => {
...
+  })
+
+  fastify.addHook('onRequest', function (req, reply, next) {
+    onRequestCalled = true
+    next()
+  })
+
+  fastify.get('/', async (req, reply) => {
+    return { hello: 'world' }
+  })
```
Comment (jsumners) [inline test/basic.test.js:651]: What part of this test is showing `middie` executing during the `preHandler` phase?

Comment (mcollina) [reply test/basic.test.js:651]: The fact that it is executed _after_ the `onRequest` has fired. I know it's a bit indirect.

**Comment by dr3** [comment]
Ive made a PR to add types for this change :) https://github.com/fastify/middie/pull/127

**Pull request merged**